t/02_yaml_schedule.t

```diff
@@ -87,4 +87,44 @@ subtest 'parse_yaml_test_schedule_recursive_conditional' => sub {
     ok $modules[3] eq 'bar/test3', "Basic scheduling";
 };
 
+subtest 'merge default schedules with flows and individual schedules' => sub {
+    use scheduler;
+    use testapi 'set_var';
+
+    # positive unit tests
+
+    my $schedule = $ypp->load_file(dirname(__FILE__) . '/data/test_data_mini_default_flows.yaml');
+    my @modules;
+    set_var('YAML_SCHEDULE_DEFAULT', '/t/data/flows/default.yaml');
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_a');
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_a_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_a_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_b');
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_b_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_b_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
```

Comment:
this is more like "key present in default and schedule but not present in flows is merged" I'm not sure "individual" can be understood correctly, as you are referring to 'individual' schedule, but I guess we should refer to the keys. wdyt?

Reply:
My point of view for individual is that we have practically 3 layers in this scheduler:

- Default: This is where the keys that you can use are defined. The scheduler is only iterating though the keys found in the default schedule, so if I mistype a key in the other layers this wil be ignored.
- Flows: This is the first level of substituion. Usually defined on TestSuite or JobGroup level.
- Individual: This is what you really "create" when designing a test. Default and Flows are just like Lego bricks that you put together and get a standard framework, but here you fine tune for what you need to test.

Reply:
I got what you mean, but it is not documented, and everybody knows this file as 'schedule' or 'YAML schedule' due to previous work, so if we refers to them as default, flows and schedule files is more simple.

Reply:
> I got what you mean, but it is not documented, and everybody knows this file as 'schedule' or 'YAML schedule' due to previous work, so if we refers to them as default, flows and schedule files is more simple.

Ok, let me rework the code for the unit tests and the script, then we see.